### PR TITLE
Use tk_listen to absorb errors on the TcpListener

### PR DESCRIPTION
**Pull request by tomaka**
[The documentation of TcpListener](https://docs.rs/tokio-tcp/0.1.1/tokio_tcp/struct.TcpListener.html#errors) suggests doing what this PR does.
The `TcpListener` can sometimes produce an error which would just close the listener.

I think handling this directly in the TCP transport is the right location. There might be valid reasons why a listener produces an error, therefore the upper layers should continue to drop the listener if an error happens.

### CODE REVIEW

File: tcp-transport/src/lib.rs [hunk 1]
```diff
@@ -108,6 +111,8 @@ impl Transport for TcpConfig {
                 .map(|listener| {
                     // Pull out a stream of sockets for incoming connections
                     listener.incoming()
+                        .sleep_on_error(Duration::from_secs(1))
```
Comment (gnunicorn) [inline tcp-transport/src/lib.rs]: The examples all use 100ms, why did you opt for waiting a full second on the connection that errored?

File: tcp-transport/src/lib.rs [hunk 2]
```diff
@@ -118,10 +123,6 @@ impl Transport for TcpConfig {
                             debug!("Incoming connection from {}", addr);
                             future::ok((TcpTransStream { inner: sock }, future::ok(addr)))
                         })
-                        .map_err(|err| {
-                            debug!("Error in TCP listener: {:?}", err);
-                            err
-                        })
                 })
```
Comment (gnunicorn) [inline tcp-transport/src/lib.rs]: While on it, why not also add the `.listen(MAX_SIMULTANEOUS_CONNECTIONS)` [from the example](https://docs.rs/tk-listen/0.2.0/tk_listen/) - we could even make that configurable, putting the `TcpConfig` to good use finally.

Comment (tomaka) [reply tcp-transport/src/lib.rs]: What `listen` does is handled by the `swarm`, although the swarm doesn't have a limit to the maximum number of connections at the moment and should have one.

File: tcp-transport/src/lib.rs (hunk 1 above)
Comment (tomaka) [reply tcp-transport/src/lib.rs]: I guess this should be in the `TcpConfig`.

**Pull request merged**